src/mono/mono/mini/interp/transform.c

```diff
@@ -3724,9 +3724,7 @@ save_seq_points (TransformData *td, MonoJitInfo *jinfo)
 static void
 interp_emit_memory_barrier (TransformData *td, int kind)
 {
-#if defined(TARGET_WASM)
```

Comment:
is `mono_memory_barrier` implementation planning to change soon for wasm, or why is this ?

Reply:
I asked and this special case was a workaround for a bug, we don't need it anymore. We'll need memory barriers with threading in play anyway, I think

Reply:
`MINT_MONO_MEMORY_BARRIER` is currently a NOP on wasm because `mono_memory_barrier ()` doesn't do anything. Adding this instruction is redundant at this moment in time, it is just a small overhead.

Reply:
The WASM block in utils/mono-membar.h needs to be removed too.